Adapt code to spatstat v2

agila5 opened the pull request:
Fix #137. 

Finally, it seems that it's working. I will complete the PR as soon as `sf` 0.9.8 is on CRAN (the 0.9.7 version is [already failing](https://cran.r-project.org/web/checks/check_results_sf.html)). The only downside is that (I think) we have to add a dependency on the 0.9.8 version of `sf` or, at least, test the `sf` version in the spatstat-related code. Do you think it's a problem?

agila5 commented:
I think it's ready to be reviewed/merged. 

IMO the only problem is that it introduces a "strong" dependency on `sf >= 0.9.8` (and also an ugly startup message related to `spatstat.geom`). Hence, I have the following proposal. I am probably the only one that takes advantage of the integration between `sfnetworks` and `spatstat`. So, maybe, we could purge all `spatstat` code from `sfnetworks`, and then I will create a new (small but with a lot of dependencies) package that extends `sfnetworks`. Opinions?

luukvdmeer commented:
> I am probably the only one that takes advantage of the integration between sfnetworks and spatstat

Not sure about that! I think it is great to have integration with other packages. I was planning to implement similar integration with e.g. `spdep`. 

What is the reason we need the sf dependency? I don't see sf functions being called in your code (but might have looked over them). If we need it, I would suggest to check for the sf version in the `check_spatstat` function, rather than setting the depedency for the whole package.

Regarding the startup message, you refer to this one?

```
Registered S3 method overwritten by 'spatstat':
  method     from
  print.boxx cli 
```

I think we can call `s3_register("spatstat.linnet::as.linnet", "sfnetwork")` inside `suppressPackageStartupMessages()` to avoid that, but not sure if that is good practice.

luukvdmeer commented on R/spatstat.R:
```diff
@@ -0,0 +1,19 @@
+      ))
+    }
+  }
+}
```
Can we include this function in the `linnet.R` file (and maybe rename that file to `spatstat.R` instead?)

agila5 replied:
Fixed with the last commits!

luukvdmeer commented on R/zzz.R:
```diff
@@ -7,7 +7,13 @@
+    utils::packageVersion("spatstat") >= "2.0.0"
+
+  if (has_spatstat_2.0) {
+    s3_register("spatstat.linnet::as.linnet", "sfnetwork")
```
In think here we could use:

```
  if (has_spatstat_2.0) {
    suppressPackageStartupMessages(
      s3_register("spatstat.linnet::as.linnet", "sfnetwork")
    )
  }
```

such that now spatstat related startup messages are printed

agila5 replied:
Fixed with the last commits!

agila5 commented:
> What is the reason we need the sf dependency? I don't see sf functions being called in your code (but might have looked over them).

The method for converting `sf` objects (with `POINT` geometry) into [`ppp`](https://rdrr.io/cran/spatstat.geom/man/ppp.object.html) objects is defined [here](https://github.com/r-spatial/sf/blob/80d6a1999929cd133b1fef77eb5a72a88a56c246/R/spatstat.R#L127-L140) and used here

https://github.com/luukvdmeer/sfnetworks/blob/236318a132f1d52f8bb93a00c0d6a72c155324e9/R/spatstat.R#L54-L55

The method for converting [`psp`](https://rdrr.io/cran/spatstat.geom/man/psp.object.html) (and [`linnet`](https://rdrr.io/cran/spatstat.linnet/man/linnet.html)) objects into `sf` objects is defined [here](https://github.com/r-spatial/sf/blob/80d6a1999929cd133b1fef77eb5a72a88a56c246/R/spatstat.R#L56-L72) and used here

https://github.com/luukvdmeer/sfnetworks/blob/236318a132f1d52f8bb93a00c0d6a72c155324e9/R/sfnetwork.R#L308-L311

The problem is that the code in the `sf` package version 0.9.7 fails because it calls `spatstat` functions that were transferred to a different package. 

> If we need it, I would suggest to check for the sf version in the check_spatstat function, rather than setting the depedency for the whole package.

Agree. Changed with the last commit. 

> Regarding the startup message, you refer to this one?

Yes

> I think we can call s3_register("spatstat.linnet::as.linnet", "sfnetwork") inside suppressPackageStartupMessages() to avoid that, but not sure if that is good practice.

I added a call to `suppressPackageStartupMessages()`, feel free to remove that.
